src/lib/AirspeedValidator/AirspeedValidator.cpp

```diff
@@ -0,0 +1,277 @@
+	float wind[2];
+	_wind_estimator.get_wind(wind);
+	wind_est.windspeed_north = wind[0];
+	wind_est.windspeed_east = wind[1];
+	float wind_cov[2];
+	_wind_estimator.get_wind_var(wind_cov);
+	wind_est.variance_north = wind_cov[0];
+	wind_est.variance_east = wind_cov[1];
+	wind_est.tas_innov = _wind_estimator.get_tas_innov();
+	wind_est.tas_innov_var = _wind_estimator.get_tas_innov_var();
+	wind_est.beta_innov = _wind_estimator.get_beta_innov();
+	wind_est.beta_innov_var = _wind_estimator.get_beta_innov_var();
+	wind_est.tas_scale = _wind_estimator.get_tas_scale();
+	return wind_est;
+}
+
+void
+AirspeedValidator::update_EAS_scale()
+{
+	if (_wind_estimator_scale_estimation_on) {
+		_EAS_scale = 1.0f / math::constrain(_wind_estimator.get_tas_scale(), 0.75f, 1.25f);
+
+	} else {
+		_EAS_scale = _airspeed_scale_manual;
+	}
+
+}
+
+void
+AirspeedValidator::update_EAS_TAS(float air_pressure_pa, float air_temperature_celsius)
+{
+	_EAS = calc_EAS_from_IAS(_IAS, _EAS_scale);
+	_TAS = calc_TAS_from_EAS(_EAS, air_pressure_pa, air_temperature_celsius);
+}
+
+void
+AirspeedValidator::check_airspeed_innovation(uint64_t time_now, float estimator_status_vel_test_ratio,
+		float estimator_status_mag_test_ratio)
+{
+	// Check normalised innovation levels with requirement for continuous data and use of hysteresis
+	// to prevent false triggering.
+
+	if (_wind_estimator.get_wind_estimator_reset()) {
+		_time_wind_estimator_initialized = time_now;
+	}
+
+	/* Wait 2s after wind estiamtor was initialzed or reset until innovation checks are done.
+		TO DO: base this on actual wind states variance, not just time.*/
+	if (!_in_fixed_wing_flight || (hrt_elapsed_time(&_time_wind_estimator_initialized) < 2000_ms)) {
+		// not in a flight condition that enables use of this check, thus pass check
+		_innovations_check_failed = false;
+		_time_last_tas_pass = time_now;
+		_time_last_tas_fail = 0;
+		_airspeed_valid = true;
+		_time_last_aspd_innov_check = time_now;
+
+	} else {
+		float dt_s = math::constrain((time_now - _time_last_aspd_innov_check) / 1e6f, 0.01f, 1.0f); // limit to 1Hz...100Hz
+
+		if (dt_s < 1.0f) {
+			float tas_innov_abs = fabsf(_wind_estimator.get_tas_innov());
+
+			if (tas_innov_abs <= _tas_innov_threshold) {
+				// record pass and reset integrator used to trigger
+				_time_last_tas_pass = time_now;
+				_apsd_innov_integ_state = 0.0f;
+
+			} else {
+				// integrate exceedance
+				_apsd_innov_integ_state += dt_s * (tas_innov_abs - _tas_innov_threshold);
+			}
+
+			if ((estimator_status_vel_test_ratio < 1.0f) && (estimator_status_mag_test_ratio < 1.0f)) {
+				// nav velocity data is likely good so airspeed innovations are able to be used
+				if ((_tas_innov_integ_threshold > 0.0f) && (_apsd_innov_integ_state > _tas_innov_integ_threshold)) {
+					_time_last_tas_fail = time_now;
+				}
+			}
+
+			if (!_innovations_check_failed) {
+				_innovations_check_failed = (hrt_elapsed_time(&_time_last_tas_pass) > TAS_INNOV_FAIL_DELAY);
```

Comment:
Same here and below:
```suggestion
				_innovations_check_failed = (time_now - _time_last_tas_pass) > TAS_INNOV_FAIL_DELAY);
```